[Cloud Security] Add cloud connector name and validation

seanrathier opened the pull request:
## Summary

Adds the connector name field to the new cloud connector in Azure and AWS, with client-side validation for connector names with a 255-character limit, and ensures the validation state properly propagates to Fleet to control the save button state.

Resolves: https://github.com/elastic/security-team/issues/14283

## Changes

### New Components
- **CloudConnectorNameField**: New form field component with real-time validation
  - Required field validation
  - 255-character maximum length validation
  - Internationalized error messages
  - Comprehensive test coverage (27 test cases)
- `CloudConnectorSelector`
   - created a reusable cloud connector selector for Azure and AWS reusable forms to share
   - options header is the connector name
   - options sub header is the Role ARN or the Managed Identity

### Validation Logic
- Updated `use_cloud_connector_setup` hook to properly propagate validation state
  - `isValid: false` disables Fleet's save button when the name is invalid
  - `isValid: undefined` allows Fleet's native validation when the name is valid
  - Added trim validation to prevent whitespace-only names
- Fixed issue where validation state wasn't consistently updating the Fleet plugin


## Testing Instructions

### Manual Testing
1. Navigate to Cloud Security CSPM or Asset Inventory integration setup
2. Select "New Connection" tab
3. Enter a cloud connector name:
   - ✅ Valid: Any name 1-255 characters
   - ❌ Invalid: Empty, whitespace only, or >255 characters
4. Verify the "Save and continue" button is:
   - **Disabled** when name is invalid (shows error message)
   - **Enabled** when name is valid (no error message)
5. Test with edge cases:
   - 255 characters (should work)
   - 256 characters (should show error and disable save)
   - Unicode characters (should work within 255 char limit)
   - Special characters (should work)




## Screenshots

<img width="785" height="701" alt="Screenshot 2025-11-13 at 10 11 54 AM" src="https://github.com/user-attachments/assets/e05a1dec-9884-40b6-a748-0e3dcf422b52" />
<img width="798" height="486" alt="Screenshot 2025-11-13 at 10 12 36 AM" src="https://github.com/user-attachments/assets/bb2464c2-b369-4ea2-9919-89633f81127c" />


## Checklist

- [x] Added client-side validation for cloud connector name field
- [x] Validation properly disables Fleet save button when invalid
- [x] Added comprehensive unit tests
- [x] Updated component naming for clarity (ReusableConnectorSelect → CloudConnectorSelector)
- [x] Verified manual testing with 255 and 256 character strings


- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [x] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [x] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

seanrathier committed:
initial LLM build

seanrathier committed:
refactored rules

seanrathier committed:
reusable select connector component

seanrathier committed:
removed code smell

seanrathier committed:
removed code smell 2

seanrathier committed:
unit tests and minor code changes

seanrathier committed:
tests

seanrathier committed:
fixed truncation

seanrathier committed:
self code review and tests

kibanamachine commented:
<!-- kbn-message-context:project-deploy-job -->
Project deployments require a Github label, please add one or more of `ci:project-deploy-(elasticsearch|observability|security)` and trigger the job through the checkbox again.

seanrathier committed:
quick check fixes

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

Omolola-Akinleye commented on x-pack/platform/plugins/shared/fleet/server/services/package_policies/utils.ts:
```diff
@@ -146,3 +146,44 @@ export function canDeployCustomPackageAsAgentlessOrThrow(
+        const hasVarsToRemove = varsToRemove.some((varName) => varName in stream.vars!);
+        if (!hasVarsToRemove) {
+          return stream;
+        }
```
why are we removing vars? what are transient vars?

Omolola-Akinleye commented on x-pack/platform/plugins/shared/fleet/server/services/package_policies/utils.test.ts:
```diff
@@ -253,3 +254,144 @@ describe('canDeployAsAgentlessOrThrow', () => {
+            {
+              id: 'stream-1',
+              vars: {
+                cloud_connector_name: { value: 'my-connector', type: 'text' },
```
We need to avoid storing the `cloud_connector_name` as input var. It should stored as part of Cloud connector save object. Similar Package Policy has text field but name is store as part of package policy save object, not a input var field.

seanrathier committed:
initial LLM build

seanrathier committed:
removed the transient var

seanrathier committed:
removed the transient var

seanrathier committed:
fixed the bugs

seanrathier committed:
fixed the bugs

seanrathier committed:
new connector working

seanrathier committed:
new connector working

seanrathier committed:
all working

seanrathier committed:
all working

seanrathier committed:
updated cloud connector object

seanrathier committed:
protecting cc from delete in policy delete secrets

seanrathier committed:
copilot review changes

seanrathier committed:
fix azure and cloud connector error rollback

seanrathier committed:
added name to agentless api

seanrathier committed:
added connector list optimizations

seanrathier committed:
added connector list optimizations

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366070)
* Commit: 7d90026fa9de1a36f2f165779ebbd588b8676c6b

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366070#019ac1cd-6d23-4b7a-95a2-e8ab121a7615)

### History
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad
* :broken_heart: [Build #361060](https://buildkite.com/elastic/kibana-pull-request/builds/361060) failed c36375939d85dc0b95ed025b08a532ae4b116588
* :broken_heart: [Build #360799](https://buildkite.com/elastic/kibana-pull-request/builds/360799) failed 1394adf1b0965bde2c5cb205a428a4a267edf060

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361195","number":361195,"commit":"5ea4d2fd14a7513156eb93d880b18b0fe49385ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361060","number":361060,"commit":"c36375939d85dc0b95ed025b08a532ae4b116588"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360799","number":360799,"commit":"1394adf1b0965bde2c5cb205a428a4a267edf060"}],"number":366070}
buildkite-pr-comment-->

Bamieh committed:
[Data Streams][Workflows] Migrate Workflows logs to data stream (#242489)

Migrates workflows logs index into a data stream using the core kbn data
stream client.

Part of this migration is ensuring efficiant initialization of the data
streams so when the data stream host plugin is disabled or behind a
feature flag then we can skip initializing the data stream, if the data
stream was already initialized in the apst the client will ensure it
undergoes migration even if the data stream will not be used.

I have also refactored the mappings helpers into a separate package to
bring type safety and utility while defining ES schemas in the client.
this brings automatic typing to the schema types and the ES doc.

Fixed Timeline output to pull from the logs, previously it was always an
empty log
<img width="1714" height="528" alt="image"
src="https://github.com/user-attachments/assets/9c158639-4133-4691-b75a-99ad0ce89224"
/>

Closes https://github.com/elastic/kibana-team/issues/2108
Closes https://github.com/elastic/kibana-team/issues/2121
Closes https://github.com/elastic/kibana/issues/243590

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366096)
* Commit: d9cebe69847462ad5ef82b1492c96eb6256671bd

### Failed CI Steps
* [Store Cache for build](https://buildkite.com/elastic/kibana-pull-request/builds/366096#019ac1f8-2d1a-4213-97f4-f3a0f73d8568)
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366096#019ac1f8-2d20-4157-b766-01ae81bc238b)

### History
* :broken_heart: [Build #366070](https://buildkite.com/elastic/kibana-pull-request/builds/366070) failed 7d90026fa9de1a36f2f165779ebbd588b8676c6b
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad
* :broken_heart: [Build #361060](https://buildkite.com/elastic/kibana-pull-request/builds/361060) failed c36375939d85dc0b95ed025b08a532ae4b116588
* :broken_heart: [Build #360799](https://buildkite.com/elastic/kibana-pull-request/builds/360799) failed 1394adf1b0965bde2c5cb205a428a4a267edf060

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366096","number":366096,"commit":"d9cebe69847462ad5ef82b1492c96eb6256671bd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361195","number":361195,"commit":"5ea4d2fd14a7513156eb93d880b18b0fe49385ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361060","number":361060,"commit":"c36375939d85dc0b95ed025b08a532ae4b116588"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360799","number":360799,"commit":"1394adf1b0965bde2c5cb205a428a4a267edf060"}],"number":366096}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366109)
* Commit: c30a40f668125c92951b23b21f2c8f9aacae4199

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366109#019ac215-4588-4b9e-ace3-61d5e0ea25ea)

### History
* :broken_heart: [Build #366096](https://buildkite.com/elastic/kibana-pull-request/builds/366096) failed d9cebe69847462ad5ef82b1492c96eb6256671bd
* :broken_heart: [Build #366070](https://buildkite.com/elastic/kibana-pull-request/builds/366070) failed 7d90026fa9de1a36f2f165779ebbd588b8676c6b
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad
* :broken_heart: [Build #361060](https://buildkite.com/elastic/kibana-pull-request/builds/361060) failed c36375939d85dc0b95ed025b08a532ae4b116588
* :broken_heart: [Build #360799](https://buildkite.com/elastic/kibana-pull-request/builds/360799) failed 1394adf1b0965bde2c5cb205a428a4a267edf060

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366109","number":366109,"commit":"c30a40f668125c92951b23b21f2c8f9aacae4199"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366096","number":366096,"commit":"d9cebe69847462ad5ef82b1492c96eb6256671bd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361195","number":361195,"commit":"5ea4d2fd14a7513156eb93d880b18b0fe49385ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361060","number":361060,"commit":"c36375939d85dc0b95ed025b08a532ae4b116588"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360799","number":360799,"commit":"1394adf1b0965bde2c5cb205a428a4a267edf060"}],"number":366109}
buildkite-pr-comment-->

seanrathier committed:
matrix working again

seanrathier committed:
typos

seanrathier committed:
remove old test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366138)
* Commit: a27209542a5a86a862665ed161a2fd45a75df60c

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366138#019ac30c-0464-414c-9052-e15b1edd90c9)

### History
* :broken_heart: [Build #366109](https://buildkite.com/elastic/kibana-pull-request/builds/366109) failed c30a40f668125c92951b23b21f2c8f9aacae4199
* :broken_heart: [Build #366096](https://buildkite.com/elastic/kibana-pull-request/builds/366096) failed d9cebe69847462ad5ef82b1492c96eb6256671bd
* :broken_heart: [Build #366070](https://buildkite.com/elastic/kibana-pull-request/builds/366070) failed 7d90026fa9de1a36f2f165779ebbd588b8676c6b
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad
* :broken_heart: [Build #361060](https://buildkite.com/elastic/kibana-pull-request/builds/361060) failed c36375939d85dc0b95ed025b08a532ae4b116588
* :broken_heart: [Build #360799](https://buildkite.com/elastic/kibana-pull-request/builds/360799) failed 1394adf1b0965bde2c5cb205a428a4a267edf060

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366138","number":366138,"commit":"a27209542a5a86a862665ed161a2fd45a75df60c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366109","number":366109,"commit":"c30a40f668125c92951b23b21f2c8f9aacae4199"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366096","number":366096,"commit":"d9cebe69847462ad5ef82b1492c96eb6256671bd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361195","number":361195,"commit":"5ea4d2fd14a7513156eb93d880b18b0fe49385ad"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361060","number":361060,"commit":"c36375939d85dc0b95ed025b08a532ae4b116588"}],"number":366138}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366140)
* Commit: dc5eea9ffcac76b5a009da56fc21f60b882968da

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366140#019ac316-bafd-4afa-88e0-812f9df6ab87)

### History
* :broken_heart: [Build #366138](https://buildkite.com/elastic/kibana-pull-request/builds/366138) failed a27209542a5a86a862665ed161a2fd45a75df60c
* :broken_heart: [Build #366109](https://buildkite.com/elastic/kibana-pull-request/builds/366109) failed c30a40f668125c92951b23b21f2c8f9aacae4199
* :broken_heart: [Build #366096](https://buildkite.com/elastic/kibana-pull-request/builds/366096) failed d9cebe69847462ad5ef82b1492c96eb6256671bd
* :broken_heart: [Build #366070](https://buildkite.com/elastic/kibana-pull-request/builds/366070) failed 7d90026fa9de1a36f2f165779ebbd588b8676c6b
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad
* :broken_heart: [Build #361060](https://buildkite.com/elastic/kibana-pull-request/builds/361060) failed c36375939d85dc0b95ed025b08a532ae4b116588

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366140","number":366140,"commit":"dc5eea9ffcac76b5a009da56fc21f60b882968da"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366138","number":366138,"commit":"a27209542a5a86a862665ed161a2fd45a75df60c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366109","number":366109,"commit":"c30a40f668125c92951b23b21f2c8f9aacae4199"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366096","number":366096,"commit":"d9cebe69847462ad5ef82b1492c96eb6256671bd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361195","number":361195,"commit":"5ea4d2fd14a7513156eb93d880b18b0fe49385ad"}],"number":366140}
buildkite-pr-comment-->

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366142)
* Commit: e5227363e6b89669f6c7eacd6ecd08de202f8e57

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/366142#019ac322-8973-4aa0-8936-f3b843286703)

### History
* :broken_heart: [Build #366140](https://buildkite.com/elastic/kibana-pull-request/builds/366140) failed dc5eea9ffcac76b5a009da56fc21f60b882968da
* :broken_heart: [Build #366138](https://buildkite.com/elastic/kibana-pull-request/builds/366138) failed a27209542a5a86a862665ed161a2fd45a75df60c
* :broken_heart: [Build #366109](https://buildkite.com/elastic/kibana-pull-request/builds/366109) failed c30a40f668125c92951b23b21f2c8f9aacae4199
* :broken_heart: [Build #366096](https://buildkite.com/elastic/kibana-pull-request/builds/366096) failed d9cebe69847462ad5ef82b1492c96eb6256671bd
* :broken_heart: [Build #366070](https://buildkite.com/elastic/kibana-pull-request/builds/366070) failed 7d90026fa9de1a36f2f165779ebbd588b8676c6b
* :broken_heart: [Build #361195](https://buildkite.com/elastic/kibana-pull-request/builds/361195) failed 5ea4d2fd14a7513156eb93d880b18b0fe49385ad

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366142","number":366142,"commit":"e5227363e6b89669f6c7eacd6ecd08de202f8e57"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366140","number":366140,"commit":"dc5eea9ffcac76b5a009da56fc21f60b882968da"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366138","number":366138,"commit":"a27209542a5a86a862665ed161a2fd45a75df60c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366109","number":366109,"commit":"c30a40f668125c92951b23b21f2c8f9aacae4199"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366096","number":366096,"commit":"d9cebe69847462ad5ef82b1492c96eb6256671bd"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366070","number":366070,"commit":"7d90026fa9de1a36f2f165779ebbd588b8676c6b"}],"number":366142}
buildkite-pr-comment-->

seanrathier committed:
check types

kibanamachine committed:
Changes from node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window   --include-path /api/agent_builder --update

kibanamachine committed:
Changes from make api-docs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/366429)
* Commit: e858f5610aa86c7a5f349a8bf0f708148d5e5cc3
* [Cloud Deployment](https://kibana-pr-242460-fe6a1c.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #366143](https://buildkite.com/elastic/kibana-pull-request/builds/366143) failed 43998b610fe7ad35ec13d4900132af33a5df42f3
* :broken_heart: [Build #366142](https://buildkite.com/elastic/kibana-pull-request/builds/366142) failed e5227363e6b89669f6c7eacd6ecd08de202f8e57
* :broken_heart: [Build #366140](https://buildkite.com/elastic/kibana-pull-request/builds/366140) failed dc5eea9ffcac76b5a009da56fc21f60b882968da
* :broken_heart: [Build #366138](https://buildkite.com/elastic/kibana-pull-request/builds/366138) failed a27209542a5a86a862665ed161a2fd45a75df60c
* :broken_heart: [Build #366109](https://buildkite.com/elastic/kibana-pull-request/builds/366109) failed c30a40f668125c92951b23b21f2c8f9aacae4199

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366429","number":366429,"commit":"e858f5610aa86c7a5f349a8bf0f708148d5e5cc3"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366143","number":366143,"commit":"43998b610fe7ad35ec13d4900132af33a5df42f3"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366142","number":366142,"commit":"e5227363e6b89669f6c7eacd6ecd08de202f8e57"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366140","number":366140,"commit":"dc5eea9ffcac76b5a009da56fc21f60b882968da"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366138","number":366138,"commit":"a27209542a5a86a862665ed161a2fd45a75df60c"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366109","number":366109,"commit":"c30a40f668125c92951b23b21f2c8f9aacae4199"}],"number":366429}
buildkite-pr-comment-->

seanrathier committed:
unit tests

seanrathier committed:
unit test fixes

seanrathier committed:
unit test fixes

nchaulet reviewed (approved):
code LGTM 🚀

opauloh reviewed (changes requested):
it seems we can bypass the validation by updating the description in an invalid state:

https://github.com/user-attachments/assets/0e4d4c74-5688-4510-8c69-74cb48b4e424


the connector and the integration was saved with 257 characters:

<img width="887" height="301" alt="image" src="https://github.com/user-attachments/assets/11081f46-37a3-4f26-91b3-9ff7c57bf9db" />


@seanrathier can we get server-side validation in the fleet API? `POST /api/fleet/agentless_policies`?

seanrathier committed:
cloud connector name backend validation

seanrathier commented:
> it seems we can bypass the validation by updating the description in an invalid state:

@opauloh  I added the backend validation. Thanks for pointing this out.

Tested with both the `xpack.fleet.enableExperimental: ['agentlessPoliciesAPI', 'useAgentlessAPIInUI']` on and off

Deployed environments now have this FF enabled.

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/367866)
* Commit: d097eff27eb390444b4ea581517751df94511c64

### Failed CI Steps
* [:pipeline::magic_wand: Run Dynamic Pipeline: .buildkite/scripts/pipelines/pull_request/pipeline.sh](https://buildkite.com/elastic/kibana-pull-request/builds/367866#019ae01d-7501-42bb-8b05-9782754cf546)

### History
* :broken_heart: [Build #367814](https://buildkite.com/elastic/kibana-pull-request/builds/367814) failed 8422d4c633a0fc5acb630f3cde3fa81f9dc8e532
* :broken_heart: [Build #367757](https://buildkite.com/elastic/kibana-pull-request/builds/367757) failed ece013316ecf6dec1e86f603640614a136422ce8
* :green_heart: [Build #367395](https://buildkite.com/elastic/kibana-pull-request/builds/367395) succeeded ce93ff3774fd90663b2461379366aa3ca58fe02f
* :yellow_heart: [Build #367336](https://buildkite.com/elastic/kibana-pull-request/builds/367336) was flaky 42d8439d4416cae8662b91a63f1ba348b68df26a
* :broken_heart: [Build #367231](https://buildkite.com/elastic/kibana-pull-request/builds/367231) failed 5346bf13ee7bb605682a797ec00d076b23e76395

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367866","number":367866,"commit":"d097eff27eb390444b4ea581517751df94511c64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367814","number":367814,"commit":"8422d4c633a0fc5acb630f3cde3fa81f9dc8e532"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367757","number":367757,"commit":"ece013316ecf6dec1e86f603640614a136422ce8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367395","number":367395,"commit":"ce93ff3774fd90663b2461379366aa3ca58fe02f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367336","number":367336,"commit":"42d8439d4416cae8662b91a63f1ba348b68df26a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367231","number":367231,"commit":"5346bf13ee7bb605682a797ec00d076b23e76395"}],"number":367866}
buildkite-pr-comment-->

seanrathier commented:
/ci

seanrathier commented:
/ci

opauloh reviewed (approved):
LGTM - thanks for adding backend validation @seanrathier, it prevents the issue with bad naming if the ui validation is bypassed and also validates when using the API directly

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/368419)
* Commit: 1615e1a63420b84a8bec297ff1e62adcde09f4d4
* [Cloud Deployment](https://kibana-pr-242460-8fb095.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #368051](https://buildkite.com/elastic/kibana-pull-request/builds/368051) failed f42a044b5bc4d6dacf7c37de34c91a7c21024685
* :broken_heart: [Build #367935](https://buildkite.com/elastic/kibana-pull-request/builds/367935) failed 917c9547d4d4e56ab784ea4e12b8edd93076617e
* :broken_heart: [Build #367894](https://buildkite.com/elastic/kibana-pull-request/builds/367894) failed d097eff27eb390444b4ea581517751df94511c64
* :broken_heart: [Build #367866](https://buildkite.com/elastic/kibana-pull-request/builds/367866) failed d097eff27eb390444b4ea581517751df94511c64
* :broken_heart: [Build #367814](https://buildkite.com/elastic/kibana-pull-request/builds/367814) failed 8422d4c633a0fc5acb630f3cde3fa81f9dc8e532

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368419","number":368419,"commit":"1615e1a63420b84a8bec297ff1e62adcde09f4d4"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/368051","number":368051,"commit":"f42a044b5bc4d6dacf7c37de34c91a7c21024685"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367935","number":367935,"commit":"917c9547d4d4e56ab784ea4e12b8edd93076617e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367894","number":367894,"commit":"d097eff27eb390444b4ea581517751df94511c64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367866","number":367866,"commit":"d097eff27eb390444b4ea581517751df94511c64"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/367814","number":367814,"commit":"8422d4c633a0fc5acb630f3cde3fa81f9dc8e532"}],"number":368419}
buildkite-pr-comment-->

seanrathier committed:
revert

seanrathier committed:
revert

seanrathier committed:
Merge branch 'main' into cloud-connector-name

kibanamachine committed:
Changes from node scripts/capture_oas_snapshot   --include-path /api/status   --include-path /api/alerting/rule/   --include-path /api/alerting/rules   --include-path /api/actions   --include-path /api/security/role   --include-path /api/spaces   --include-path /api/streams   --include-path /api/fleet   --include-path /api/saved_objects/_import   --include-path /api/saved_objects/_export   --include-path /api/maintenance_window   --include-path /api/agent_builder --update

kibanamachine committed:
Changes from make api-docs
